casper/src/main/scala/coop/rchain/casper/util/DagOperations.scala

```diff
@@ -91,37 +91,80 @@ object DagOperations {
     })
   }
 
+  private val blockMetadataByNumDecreasing: Ordering[BlockMetadata] =
+    (l: BlockMetadata, r: BlockMetadata) => {
+      def compareByteString(l: ByteString, r: ByteString): Int =
+        l.hashCode().compareTo(r.hashCode())
```

Comment:
I know you're not the author, but I find this a terrible hack:

1. given two byte strings, it's very unclear which one is going to be the 'smaller' one, as their hash is used for comparison,
2. this isn't always a tie breaker, as hashes can collide as well.

We should compare the byte strings lexicographically.

Reply:
Replaced this with basically `Ordering.Iterable[Byte]`